# Las Fénix Sobre Ruedas
## 🐼 Organizando el Escuadrón con PandasMisión 02: Organizando el Escuadrón con Pandas

### 📝 **MEMORÁNDUM CONFIDENCIAL**


**PARA:** Nuestra Analista de Datos Estrella

**DE:** Isabel "La Centella" Montero, Entrenadora Principal, Las Fénix Sobre Ruedas

**ASUNTO:** Necesito Claridad en el Equipo... ¡Y Detectar Patrones!

Analista,

El trabajo con NumPy nos dio una primera visión de los números, pero ahora necesito que demos un paso más allá. Quiero que uses **Pandas** para organizar toda la información en una estructura clara, como una tabla de mandos. Esto nos permitirá ver no solo las cifras, sino también los nombres, las fechas, los comentarios... todo en su sitio. Pandas es una biblioteca de Python especializada en el manejo y análisis de estructuras de datos.

He estado revisando los informes preliminares y, aunque los números generales del equipo son aceptables, tengo la sensación de que hay **algunas dinámicas internas y rendimientos individuales que quiero que empecemos a observar con más detalle**. Es importante que te familiarices bien con las herramientas de selección y filtrado de Pandas.

**Tu misión con Pandas (el DataFrame ya estará cargado para ti):**
1.  **Inspección General:** Familiarízate con el DataFrame `df_fenix` que ya hemos cargado: dimensiones (`.shape`), información general (`.info()`), primeras/últimas filas (`.head()`, `.tail()`), nombres de columnas (`.columns`), tipos de datos (`.dtypes`).
2.  **Tipos de Datos Adecuados:** Asegúrate de que cada columna tenga el tipo de dato correcto. Presta especial atención a `Fecha_Sesion` (a `datetime`) y a las columnas numéricas que puedan estar como `object`. Límpialas y conviértelas a numéricas.
3.  **Selección de Columnas y Filas:** Practica la selección de datos usando los nombres de columna y los métodos `iloc` y `loc`.
4.  **Filtrado por Condiciones:** Realiza algunos filtros básicos para seleccionar sesiones que cumplan ciertos criterios.
5.  **Nuevas Perspectivas:** Crea la columna `Faltas_Totales`.
6.  **Análisis Enfocado (¡Aquí empieza lo interesante!):** Una vez te hayas familiarizado con lo anterior, quiero que apliques esas habilidades de filtrado para examinar más de cerca a "EstrellaFugaz77" y "NovaProdigy21". Compara sus puntuaciones medias en sesiones clave y revisa sus comentarios.
7.  **Informe de Inteligencia Pandas:** Al final, resume los pasos de limpieza y cualquier observación inicial, especialmente sobre esas dos patinadoras.

Tenemos que empezar a entender no solo *qué* pasa, sino a *quién* le pasa y *por qué*.

In [1]:
import pandas as pd
import numpy as np # Lo podríamos necesitar para np.nan o alguna conversión

# Configuración de Pandas
pd.set_option('display.max_columns', None)
pd.set_option('display.max_colwidth', None)
pd.set_option('display.float_format', lambda x: '%.2f' % x) # Formatear floats a 2 decimales

In [2]:
# ¡Atención Analista! La Centella te ha facilitado este primer paso.
# El DataFrame 'rendimiento_fenix.csv' ya está cargado para ti en la variable df_fenix.
# ¡Directo a la acción!

try:
    df_fenix = pd.read_csv('rendimiento_fenix.csv', sep=';') 
    print("DataFrame 'rendimiento_fenix.csv' cargado exitosamente en la variable 'df_fenix'.")
    print(f"Dimensiones: {df_fenix.shape[0]} filas y {df_fenix.shape[1]} columnas.") 
except FileNotFoundError:
    print("Error CRÍTICO: No se encontró el archivo 'rendimiento_fenix.csv'.")
    print("Asegúrate de haber ejecutado 'generador_csv_fenix_v2.py' (y renombrado si es necesario a 'rendimiento_fenix.csv') y que el archivo esté en la ubicación correcta.")
    df_fenix = pd.DataFrame() # Crear un DataFrame vacío para evitar errores si no se carga

DataFrame 'rendimiento_fenix.csv' cargado exitosamente en la variable 'df_fenix'.
Dimensiones: 200 filas y 21 columnas.


In [3]:
df_fenix

ID_Patinadora  Alias_Patinadora         Fecha_Sesion  \
0        FENIX_11          ZigZag45  2024-08-27 19:00:00   
1        FENIX_05        Dinamita88  2024-08-03 10:45:00   
2        FENIX_09     NovaProdigy21           17/09/2024   
3        FENIX_07  FuriaTranquila61            16-Sep-24   
4        FENIX_11          ZigZag45  2024-09-28 15:15:00   
..            ...               ...                  ...   
195      FENIX_07  FuriaTranquila61  2024-08-25 14:30:00   
196      FENIX_07  FuriaTranquila61           09/08/2024   
197      FENIX_06       AceroPuro97            03-Oct-24   
198      FENIX_04      Torbellino63            18-Oct-24   
199      FENIX_01        Dinamita27           21/08/2024   

                   Tipo_Sesion Duracion_Min Pulsaciones_Max_BPM  \
0        Simulacro Competición          164                 177   
1        Simulacro Competición           55                 162   
2       Clasificatoria Interna    Pendiente                 182   
3        Simulacro Competición           2h              183bpm   
4                  Desconocido          107                 194   
..                         ...          ...                 ...   
195  Técnica Específica Saltos          138                 192   
196    Técnica Específica Pies      82 min.                 175   
197  Técnica Específica Saltos           NR                 196   
198                Comp. Local           54                 175   
199                        NaN           59                 190   

    Pulsaciones_Media_BPM Puntuacion_Rendimiento Nivel_Cansancio_Post  \
0                     148                    9.5                    5   
1                     142                    7.3                    3   
2                     158                regular                    4   
3                  163bpm                    8.5                  4 U   
4                     174                    8.4                  2 U   
..                    ...                    ...                  ...   
195                   164                    9.0                    5   
196                   153                    6.5                  5 U   
197                   181                    8.8                    1   
198                   157                    9.3                    5   
199                   169                    9.2                    5   

    Horas_Sueño_Previas             Comentario_Patinadora  \
0                 Error                               NaN   
1                   6.9                         Pendiente   
2                   7.5                               NaN   
3                   8.8  No me sentí cómoda con las botas   
4                   6,2                               NaN   
..                  ...                               ...   
195                 8.9  No me sentí cómoda con las botas   
196                 7.1                               NaN   
197                 7.5                               NaN   
198                 8.1                               NaN   
199                 6.1                               NaN   

            Comentario_Entrenadora Equipamiento_Principal_Test  \
0                              NaN            BOTAS_AIRFLOW_V1   
1                              NaN                         NaN   
2    Excelente progresión en giros       RUEDAS_SPEEDDEMON_92A   
3                              NaN          RUEDAS_GRIPMAX_88A   
4                              NaN                Ruedas Rojas   
..                             ...                         ...   
195              Actitud impecable      PAT_MOD_TITANIUM_R_PRO   
196             Falta explosividad  PAT_MOD_TITAINUM_R_PRO_EXP   
197   Necesita más trabajo de base          RUEDAS_GRIPMAX_88A   
198                            NaN      PAT_MOD_TITANIUM_R_PRO   
199  Excelente progresión en giros      PAT_MOD_TITANIUM_R_PRO   

    Numero_Caidas_Leves Numero_Fallos_Tecnicos Rival_Observada_ID  \
0     

### 🕵️‍♀️ Tarea 1: Inspección General del DataFrame ⚙️

Ahora que `df_fenix` está cargado, vamos a conocer nuestros datos.

"La Centella" quiere un vistazo general. Vamos a usar algunos métodos básicos de Pandas para entender la estructura y contenido de nuestro DataFrame.

**Tu tarea:**
* Muestra las primeras 3 filas.
* Muestra las últimas 3 filas.
* Obtén el nombre de las columnas
* Obtén información general del DataFrame (tipos de datos, nulos).
* Genera estadísticas descriptivas para las columnas numéricas.

In [4]:
if not df_fenix.empty:
    print("-"*60)
    print("3 primeras filas del DF:")
    print("-"*60)
    display(df_fenix.head(3))
    print("-"*60)
    print("3 primeras últimas del DF:")
    print("-"*60)
    display(df_fenix.tail(3))
    print("-"*60)
    print("Nombres de las columnas:")
    print("-"*60)
    display(df_fenix.columns.tolist())
    print("-"*60)
    print("Información general del DF:")
    print("-"*60)
    display(df_fenix.info())
    print("-"*60)
    print("Estadisticas descriptivas del DF:")
    print("-"*60)
    display(df_fenix.describe())
else:
    print("El DataFrame 'df_fenix' está vacío. No se puede realizar la inspección.")

------------------------------------------------------------
3 primeras filas del DF:
------------------------------------------------------------


ID_Patinadora Alias_Patinadora         Fecha_Sesion             Tipo_Sesion  \
0      FENIX_11         ZigZag45  2024-08-27 19:00:00   Simulacro Competición   
1      FENIX_05       Dinamita88  2024-08-03 10:45:00   Simulacro Competición   
2      FENIX_09    NovaProdigy21           17/09/2024  Clasificatoria Interna   

  Duracion_Min Pulsaciones_Max_BPM Pulsaciones_Media_BPM  \
0          164                 177                   148   
1           55                 162                   142   
2    Pendiente                 182                   158   

  Puntuacion_Rendimiento Nivel_Cansancio_Post Horas_Sueño_Previas  \
0                    9.5                    5               Error   
1                    7.3                    3                 6.9   
2                regular                    4                 7.5   

  Comentario_Patinadora         Comentario_Entrenadora  \
0                   NaN                            NaN   
1             Pendiente                            NaN   
2                   NaN  Excelente progresión en giros   

  Equipamiento_Principal_Test Numero_Caidas_Leves Numero_Fallos_Tecnicos  \
0            BOTAS_AIRFLOW_V1                   0                      1   
1                         NaN                   2                      0   
2       RUEDAS_SPEEDDEMON_92A                   0              bastantes   

  Rival_Observada_ID Foco_Mental_Reportado Lesion_Reportada_S_N  \
0                NaN                     3                 Leve   
1                NaN                normal                    N   
2                NaN                     5                    N   

  Zona_Entrenamiento_Codigo Suplemento_Autorizado_Consumido  \
0           Pista Principal       Barrita Proteica Almendra   
1          Pista PPrincipal       Barrita Proteica Almendra   
2                   PISTA_A            rCeatina Monohidrato   

  Objetivo_Principal_Sesion  
0         Resistencia Fondo  
1                       NaN  
2      Fuerza Tren Inferior

------------------------------------------------------------
3 primeras últimas del DF:
------------------------------------------------------------


ID_Patinadora Alias_Patinadora Fecha_Sesion                Tipo_Sesion  \
197      FENIX_06      AceroPuro97    03-Oct-24  Técnica Específica Saltos   
198      FENIX_04     Torbellino63    18-Oct-24                Comp. Local   
199      FENIX_01       Dinamita27   21/08/2024                        NaN   

    Duracion_Min Pulsaciones_Max_BPM Pulsaciones_Media_BPM  \
197           NR                 196                   181   
198           54                 175                   157   
199           59                 190                   169   

    Puntuacion_Rendimiento Nivel_Cansancio_Post Horas_Sueño_Previas  \
197                    8.8                    1                 7.5   
198                    9.3                    5                 8.1   
199                    9.2                    5                 6.1   

    Comentario_Patinadora         Comentario_Entrenadora  \
197                   NaN   Necesita más trabajo de base   
198                   NaN                            NaN   
199                   NaN  Excelente progresión en giros   

    Equipamiento_Principal_Test Numero_Caidas_Leves Numero_Fallos_Tecnicos  \
197          RUEDAS_GRIPMAX_88A                   3                    0 U   
198      PAT_MOD_TITANIUM_R_PRO                   0                      0   
199      PAT_MOD_TITANIUM_R_PRO                   0                      0   

    Rival_Observada_ID Foco_Mental_Reportado Lesion_Reportada_S_N  \
197  RIVAL_DRAGONES_04                     4                    N   
198                NaN                     5                    N   
199                NaN                     4                LLeve   

    Zona_Entrenamiento_Codigo Suplemento_Autorizado_Consumido  \
197                   PISTA_B                              NR   
198                       NaN                     Desconocido   
199                       GYM                       Pendiente   

    Objetivo_Principal_Sesion  
197         Coreografia Grupo  
198         Resistencia Fondo  
199            Velocidad Max.

------------------------------------------------------------
Nombres de las columnas:
------------------------------------------------------------


['ID_Patinadora',
 'Alias_Patinadora',
 'Fecha_Sesion',
 'Tipo_Sesion',
 'Duracion_Min',
 'Pulsaciones_Max_BPM',
 'Pulsaciones_Media_BPM',
 'Puntuacion_Rendimiento',
 'Nivel_Cansancio_Post',
 'Horas_Sueño_Previas',
 'Comentario_Patinadora',
 'Comentario_Entrenadora',
 'Equipamiento_Principal_Test',
 'Numero_Caidas_Leves',
 'Numero_Fallos_Tecnicos',
 'Rival_Observada_ID',
 'Foco_Mental_Reportado',
 'Lesion_Reportada_S_N',
 'Zona_Entrenamiento_Codigo',
 'Suplemento_Autorizado_Consumido',
 'Objetivo_Principal_Sesion']

------------------------------------------------------------
Información general del DF:
------------------------------------------------------------
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 200 entries, 0 to 199
Data columns (total 21 columns):
 #   Column                           Non-Null Count  Dtype 
---  ------                           --------------  ----- 
 0   ID_Patinadora                    200 non-null    object
 1   Alias_Patinadora                 199 non-null    object
 2   Fecha_Sesion                     197 non-null    object
 3   Tipo_Sesion                      192 non-null    object
 4   Duracion_Min                     195 non-null    object
 5   Pulsaciones_Max_BPM              196 non-null    object
 6   Pulsaciones_Media_BPM            196 non-null    object
 7   Puntuacion_Rendimiento           196 non-null    object
 8   Nivel_Cansancio_Post             194 non-null    object
 9   Horas_Sueño_Previas              197 non-null    object
 10  Comentar

None

------------------------------------------------------------
Estadisticas descriptivas del DF:
------------------------------------------------------------


ID_Patinadora Alias_Patinadora Fecha_Sesion            Tipo_Sesion  \
count            200              199          197                    192   
unique            15               28          162                     32   
top         FENIX_13      AceroPuro21   15/10/2024  Simulacro Competición   
freq              17               17            3                     22   

       Duracion_Min Pulsaciones_Max_BPM Pulsaciones_Media_BPM  \
count           195                 196                   196   
unique          127                  60                    72   
top              1h                 187                   156   
freq              7                   8                     9   

       Puntuacion_Rendimiento Nivel_Cansancio_Post Horas_Sueño_Previas  \
count                     196                  194                 197   
unique                     68                   16                  80   
top                 excelente                    4               siete   
freq                       11                   42                  12   

                   Comentario_Patinadora Comentario_Entrenadora  \
count                                 57                     74   
unique                                14                     14   
top     No me sentí cómoda con las botas              Potencial   
freq                                  11                     12   

       Equipamiento_Principal_Test Numero_Caidas_Leves Numero_Fallos_Tecnicos  \
count                          195                 192                    197   
unique                          36                  14                     17   
top               BOTAS_AIRFLOW_V1                   0                      0   
freq                            31                  84                     73   

       Rival_Observada_ID Foco_Mental_Reportado Lesion_Reportada_S_N  \
count                  48                   194                  200   
unique                 24                    16                    8   
top     RIVAL_HALCONES_03                     4                    N   
freq                    5                    37                  157   

       Zona_Entrenamiento_Codigo Suplemento_Autorizado_Consumido  \
count                        197                             187   
unique                        28                              33   
top                          GYM            Creatina Monohidrato   
freq                          39                              28   

       Objetivo_Principal_Sesion  
count                        197  
unique                        42  
top                 Flexibilidad  
freq                          22

------------------------------------------------------------------------------------------------------------------------------------------------------------------

`📝 **Documentación del ejercicio por parte de la alumna**📝:`

**
**

------------------------------------------------------------------------------------------------------------------------------------------------------------------

### 🏷️ (Opcional) Tarea Adicional: Renombrado Estratégico de Columnas

Revisa los nombres de las columnas. Si alguna necesita un nombre más claro o práctico para trabajar, este es el momento de usar `.rename()`.

*Para este dataset, los nombres deberían ser bastante descriptivos. Realiza esta tarea solo si identificas alguna columna que se beneficiaría de un cambio.*

In [5]:
# Renombro la columna "Suplemento_Autorizado_Consumido" por "Suplementacion_Autorizada"

df_fenix.rename(columns={"Suplemento_Autorizado_Consumido":"Suplementacion_Autorizada"}, inplace=True)
print(df_fenix.columns.tolist())

['ID_Patinadora', 'Alias_Patinadora', 'Fecha_Sesion', 'Tipo_Sesion', 'Duracion_Min', 'Pulsaciones_Max_BPM', 'Pulsaciones_Media_BPM', 'Puntuacion_Rendimiento', 'Nivel_Cansancio_Post', 'Horas_Sueño_Previas', 'Comentario_Patinadora', 'Comentario_Entrenadora', 'Equipamiento_Principal_Test', 'Numero_Caidas_Leves', 'Numero_Fallos_Tecnicos', 'Rival_Observada_ID', 'Foco_Mental_Reportado', 'Lesion_Reportada_S_N', 'Zona_Entrenamiento_Codigo', 'Suplementacion_Autorizada', 'Objetivo_Principal_Sesion']


------------------------------------------------------------------------------------------------------------------------------------------------------------------

`📝 **Documentación del ejercicio por parte de la alumna**📝:`

**
**

------------------------------------------------------------------------------------------------------------------------------------------------------------------

### 🔧 Tarea 2: Asegurando los Tipos de Datos Correctos

Es crucial que cada columna tenga el tipo de dato apropiado para poder realizar análisis correctos. Pandas es flexible y puede manejar una amplia variedad de formatos de datos [661].

- `Fecha_Sesion` debe ser de tipo `datetime`. (Es probable que con el batiburrillo que tenemos montado la funcion pd.to_datetime sin más no sea suficiente, pero puedes crear una función teniendo en cuenta esto:

        - Formatos_a_probar = [
                    '%Y-%m-%d %H:%M:%S',  # Formato estándar: '2024-08-27 19:00:00'
                    '%d-%b-%y',          # Formato clave: '16-Sep-24'
                    '%d/%m/%Y',          # Formato europeo: '17/09/2024'
                    '%B %d, %Y',         # Nombre de mes completo: 'September 03, 2024'
                ])
- Columnas que deberían ser numéricas pero puedan estar como `object` (ej. `Duracion_Min`, `Pulsaciones_Max_BPM`, `Puntuacion_Rendimiento`, etc.) deben limpiarse (quitar caracteres no numéricos si es simple, como "bpm") y convertirse. Utiliza `pd.to_numeric(errors='coerce')` para manejar valores que no se puedan convertir, los cuales se transformarán en `NaN`.

In [6]:
# Convierto Fecha_Sesion en datetime (Pierdo muchisimos datos, voy a probar con array.Numpy)

""" if 'Fecha_Sesion' in df_fenix.columns:
    df_fenix['Fecha_Sesion'] = pd.to_datetime(df_fenix['Fecha_Sesion'], errors='coerce')
    if df_fenix['Fecha_Sesion'].isnull().any():
        print("Algunos datos no se han podido convertir y pasan a nulos")

display(df_fenix['Fecha_Sesion']) """


' if \'Fecha_Sesion\' in df_fenix.columns:\n    df_fenix[\'Fecha_Sesion\'] = pd.to_datetime(df_fenix[\'Fecha_Sesion\'], errors=\'coerce\')\n    if df_fenix[\'Fecha_Sesion\'].isnull().any():\n        print("Algunos datos no se han podido convertir y pasan a nulos")\n\ndisplay(df_fenix[\'Fecha_Sesion\']) '

In [7]:
from pandas import Timestamp

fechas_array = df_fenix['Fecha_Sesion'].to_numpy()

def parseador_para_datos_mixtos(valor):
    """
    Analiza un valor que puede ser un string, un timestamp ya convertido o un nulo.
    """
    # 1. Si ya es un Timestamp válido, devolverlo tal cual.
    if isinstance(valor, Timestamp) and pd.notna(valor):
        return valor

    # 2. Si es un string, intentar analizarlo con varios formatos.
    if isinstance(valor, str):
        formatos_a_probar = [
            '%Y-%m-%d %H:%M:%S',  # Formato estándar: '2024-08-27 19:00:00'
            '%d-%b-%y',          # Formato clave: '16-Sep-24'
            '%d/%m/%Y',          # Formato europeo: '17/09/2024'
            '%B %d, %Y',         # Nombre de mes completo: 'September 03, 2024'
        ]
        for fmt in formatos_a_probar:
            try:
                return pd.to_datetime(valor, format=fmt)
            except (ValueError, TypeError):
                continue
        # Último intento con el parser genérico si los formatos específicos fallan
        try:
            return pd.to_datetime(valor, dayfirst=True)
        except (ValueError, TypeError):
            return pd.NaT

    # 3. Si es cualquier otra cosa (nan, NaT inicial, etc.), es un nulo.
    return pd.NaT

# 4. Crear la Serie de pandas
s = pd.Series(fechas_array)

# 5. Aplicar la función de análisis correcta a cada elemento de la serie
fechas_limpias = s.apply(parseador_para_datos_mixtos)

# 6. Introduce el array al df
df_fenix['Fecha_Sesion'] = fechas_limpias
df_fenix['Fecha_Sesion'] = df_fenix['Fecha_Sesion'].astype('datetime64[ns]')


In [8]:
# Creo una función para la conversión numérica
def conversion_numerica(df, columna):  
    df[columna] = pd.to_numeric(df[columna], errors='coerce')
    return

In [9]:
#Creo una lista y utilizo un bucle for para convertir las columnas
columnas_numericas = ['Duracion_Min','Pulsaciones_Max_BPM','Pulsaciones_Media_BPM','Puntuacion_Rendimiento','Nivel_Cansancio_Post','Horas_Sueño_Previas','Numero_Caidas_Leves','Numero_Fallos_Tecnicos','Foco_Mental_Reportado' ]

for columna in columnas_numericas:
    print(f"Convirtiendo la columna: {columna}:")
    conversion_numerica(df_fenix, columna)

Convirtiendo la columna: Duracion_Min:
Convirtiendo la columna: Pulsaciones_Max_BPM:
Convirtiendo la columna: Pulsaciones_Media_BPM:
Convirtiendo la columna: Puntuacion_Rendimiento:
Convirtiendo la columna: Nivel_Cansancio_Post:
Convirtiendo la columna: Horas_Sueño_Previas:
Convirtiendo la columna: Numero_Caidas_Leves:
Convirtiendo la columna: Numero_Fallos_Tecnicos:
Convirtiendo la columna: Foco_Mental_Reportado:


------------------------------------------------------------------------------------------------------------------------------------------------------------------

`📝 **Documentación del ejercicio por parte de la alumna**📝:`

**
**

------------------------------------------------------------------------------------------------------------------------------------------------------------------

### 🎯 Tarea 3: Selección de Columnas y Filas (`loc` e `iloc`)

Practicaremos cómo acceder a partes específicas de nuestro DataFrame. `loc` se usa para acceder por etiquetas y `iloc` por posición numérica.

**Instrucciones:**
1.  Selecciona y muestra solo la columna `Alias_Patinadora`. ¿Qué tipo de estructura de Pandas es?
2.  Selecciona y muestra las columnas `Alias_Patinadora`, `Fecha_Sesion` y `Puntuacion_Rendimiento`.
3.  Usando `iloc`:
    * .a Selecciona y muestra la tercera fila del DataFrame (índice 2).
    * .b Selecciona y muestra las filas en las posiciones 10 a 14 (inclusive la 14).
    * .c Selecciona y muestra las primeras 3 filas y las primeras 3 columnas.
4.  Usando `loc`:
    * .a Selecciona y muestra las filas con índice (etiqueta) de 0 a 5 (inclusive la 5).
    * .b Selecciona las filas con índice 0 a 5 y solo las columnas `Alias_Patinadora` y `Puntuacion_Rendimiento`.

In [10]:
# 1. 
alias_patinadora = df_fenix['Alias_Patinadora']
display(alias_patinadora)
print("La estructura es: \n")
print(type(alias_patinadora))

0              ZigZag45
1            Dinamita88
2         NovaProdigy21
3      FuriaTranquila61
4              ZigZag45
             ...       
195    FuriaTranquila61
196    FuriaTranquila61
197         AceroPuro97
198        Torbellino63
199          Dinamita27
Name: Alias_Patinadora, Length: 200, dtype: object

La estructura es: 

<class 'pandas.core.series.Series'>


In [11]:
# 2.
serie_columnas = ['Alias_Patinadora', 'Fecha_Sesion', 'Puntuacion_Rendimiento']
df_f3 = df_fenix[serie_columnas]
display(df_f3)
print("La estructura es: \n")
print(type(df_f3))

Alias_Patinadora        Fecha_Sesion  Puntuacion_Rendimiento
0            ZigZag45 2024-08-27 19:00:00                    9.50
1          Dinamita88 2024-08-03 10:45:00                    7.30
2       NovaProdigy21 2024-09-17 00:00:00                     NaN
3    FuriaTranquila61 2024-09-16 00:00:00                    8.50
4            ZigZag45 2024-09-28 15:15:00                    8.40
..                ...                 ...                     ...
195  FuriaTranquila61 2024-08-25 14:30:00                    9.00
196  FuriaTranquila61 2024-08-09 00:00:00                    6.50
197       AceroPuro97 2024-10-03 00:00:00                    8.80
198      Torbellino63 2024-10-18 00:00:00                    9.30
199        Dinamita27 2024-08-21 00:00:00                    9.20

[200 rows x 3 columns]

La estructura es: 

<class 'pandas.core.frame.DataFrame'>


In [12]:
# 3.a
tercera_fila = df_fenix.iloc[2]
display(tercera_fila)
print("-"*150)
# 3.b
filas_10a14 = df_fenix.iloc[10:15]
display(filas_10a14)
print("-"*150)
# 3.c
tresc_tresf = df_fenix.iloc[:3,:3]
display(tresc_tresf)


ID_Patinadora                                       FENIX_09
Alias_Patinadora                               NovaProdigy21
Fecha_Sesion                             2024-09-17 00:00:00
Tipo_Sesion                           Clasificatoria Interna
Duracion_Min                                             NaN
Pulsaciones_Max_BPM                                   182.00
Pulsaciones_Media_BPM                                 158.00
Puntuacion_Rendimiento                                   NaN
Nivel_Cansancio_Post                                    4.00
Horas_Sueño_Previas                                     7.50
Comentario_Patinadora                                    NaN
Comentario_Entrenadora         Excelente progresión en giros
Equipamiento_Principal_Test            RUEDAS_SPEEDDEMON_92A
Numero_Caidas_Leves                                     0.00
Numero_Fallos_Tecnicos                                   NaN
Rival_Observada_ID                                       NaN
Foco_Mental_Reportado   

------------------------------------------------------------------------------------------------------------------------------------------------------


ID_Patinadora Alias_Patinadora        Fecha_Sesion  \
10      FENIX_08       Vendaval51 2024-09-26 00:00:00   
11      FENIX_14  CentellaNegra76 2024-09-29 00:00:00   
12      FENIX_06      AceroPuro97 2024-09-02 00:00:00   
13      FENIX_09    NovaProdigy21 2024-08-16 16:30:00   
14      FENIX_08       Vtndaval51 2024-08-29 00:00:00   

                  Tipo_Sesion  Duracion_Min  Pulsaciones_Max_BPM  \
10                        NaN         30.00               199.00   
11         Preparación Física           NaN                  NaN   
12      Entrenamiento General         87.00               193.00   
13      Simulacro Competición         84.00               178.00   
14  Técnica Específica Saltos         59.00               160.00   

    Pulsaciones_Media_BPM  Puntuacion_Rendimiento  Nivel_Cansancio_Post  \
10                 181.00                    9.30                   NaN   
11                 165.00                     NaN                  1.00   
12                 172.00                     NaN                   NaN   
13                 156.00                    9.30                   NaN   
14                    NaN                     NaN                   NaN   

    Horas_Sueño_Previas             Comentario_Patinadora  \
10                 8.10  No me sentí cómoda con las botas   
11                  NaN    Necesito mejorar mi Salto Axel   
12                 6.80                               NaN   
13                 8.00                               NaN   
14                  NaN                    Hoy todo fluyó   

   Comentario_Entrenadora Equipamiento_Principal_Test  Numero_Caidas_Leves  \
10                    NaN       PAT_MOD_CARBONX_R_STD                 3.00   
11                    NaN           BOTAS_FLEXLITE_V2                 3.00   
12                    NaN      PAT_MOD_TITANIUM_R_PRO                 0.00   
13                     NR            BOTAS_AIRFLOW_V1                 0.00   
14                    NaN       APT_MOD_CARBONX_R_STD                 3.00   

    Numero_Fallos_Tecnicos Rival_Observada_ID  Foco_Mental_Reportado  \
10                    2.00  RIVAL_HALCONES_03                   3.00   
11                     NaN                NaN                   2.00   
12                     NaN                NaN                   4.00   
13                    1.00     RIVAL_LOBAS_03                   4.00   
14                    0.00                NaN                    NaN   

   Lesion_Reportada_S_N Zona_Entrenamiento_Codigo Suplementacion_Autorizada  \
10                    N           EXTERIOR_PARQUE                 Pendiente   
11                    N                  PISTAA_B      Creatina Monohidrato   
12                    N                   PISTA_B              BCAA Manzana   
13                    N                   PISTA_B                 Pendiente   
14                    N                       GYM         Electrolitos Lima   

   Objetivo_Principal_Sesion  
10         Coreografia Grupo  
11       Recuperacion Activa  
12         Resistencia Fondo  
13              Flexibilidad  
14            Velocidad Max.

------------------------------------------------------------------------------------------------------------------------------------------------------


ID_Patinadora Alias_Patinadora        Fecha_Sesion
0      FENIX_11         ZigZag45 2024-08-27 19:00:00
1      FENIX_05       Dinamita88 2024-08-03 10:45:00
2      FENIX_09    NovaProdigy21 2024-09-17 00:00:00


4.  Usando `loc`:
    * Selecciona y muestra las filas con índice (etiqueta) de 0 a 5 (inclusive la 5).
    * Selecciona las filas con índice 0 a 5 y solo las columnas `Alias_Patinadora` y `Puntuacion_Rendimiento`.

In [13]:
# 4.a
ej_loc = df_fenix.loc[0:5]
display(ej_loc)

# 4.b 
columnas = ['Alias_Patinadora', 'Puntuacion_Rendimiento']
ej_loc_dos = df_fenix[columnas].loc[0:4]
display(ej_loc_dos)

ID_Patinadora  Alias_Patinadora        Fecha_Sesion             Tipo_Sesion  \
0      FENIX_11          ZigZag45 2024-08-27 19:00:00   Simulacro Competición   
1      FENIX_05        Dinamita88 2024-08-03 10:45:00   Simulacro Competición   
2      FENIX_09     NovaProdigy21 2024-09-17 00:00:00  Clasificatoria Interna   
3      FENIX_07  FuriaTranquila61 2024-09-16 00:00:00   Simulacro Competición   
4      FENIX_11          ZigZag45 2024-09-28 15:15:00             Desconocido   
5      FENIX_10          ZigZag97 2024-10-12 00:00:00     Rutina Coreográfica   

   Duracion_Min  Pulsaciones_Max_BPM  Pulsaciones_Media_BPM  \
0        164.00               177.00                 148.00   
1         55.00               162.00                 142.00   
2           NaN               182.00                 158.00   
3           NaN                  NaN                    NaN   
4        107.00               194.00                 174.00   
5        159.00               166.00                 143.00   

   Puntuacion_Rendimiento  Nivel_Cansancio_Post  Horas_Sueño_Previas  \
0                    9.50                  5.00                  NaN   
1                    7.30                  3.00                 6.90   
2                     NaN                  4.00                 7.50   
3                    8.50                   NaN                 8.80   
4                    8.40                   NaN                  NaN   
5                     NaN                  2.00                 7.50   

              Comentario_Patinadora         Comentario_Entrenadora  \
0                               NaN                            NaN   
1                         Pendiente                            NaN   
2                               NaN  Excelente progresión en giros   
3  No me sentí cómoda con las botas                            NaN   
4                               NaN                            NaN   
5                               NaN                  Revisar vídeo   

  Equipamiento_Principal_Test  Numero_Caidas_Leves  Numero_Fallos_Tecnicos  \
0            BOTAS_AIRFLOW_V1                 0.00                    1.00   
1                         NaN                 2.00                    0.00   
2       RUEDAS_SPEEDDEMON_92A                 0.00                     NaN   
3          RUEDAS_GRIPMAX_88A                 1.00                    0.00   
4                Ruedas Rojas                 1.00                    1.00   
5            BOTAS_AIRFLOW_V1                 0.00                    0.00   

  Rival_Observada_ID  Foco_Mental_Reportado Lesion_Reportada_S_N  \
0                NaN                   3.00                 Leve   
1                NaN                    NaN                    N   
2                NaN                   5.00                    N   
3     RIVAL_LOBAS_04                   1.00                    N   
4                NaN                   5.00                    N   
5                NaN                   1.00                    N   

  Zona_Entrenamiento_Codigo  Suplementacion_Autorizada  \
0           Pista Principal  Barrita Proteica Almendra   
1          Pista PPrincipal  Barrita Proteica Almendra   
2                   PISTA_A       rCeatina Monohidrato   
3           Pista Principal                    Ninguno   
4                   PISTA_A                         NR   
5                   PISTA_A         Electrolittos Lima   

  Objetivo_Principal_Sesion  
0         Resistencia Fondo  
1                       NaN  
2      Fuerza Tren Inferior  
3       Recuperacion Activa  
4                   General  
5             Coordinacioon

Alias_Patinadora  Puntuacion_Rendimiento
0          ZigZag45                    9.50
1        Dinamita88                    7.30
2     NovaProdigy21                     NaN
3  FuriaTranquila61                    8.50
4          ZigZag45                    8.40

------------------------------------------------------------------------------------------------------------------------------------------------------------------

`📝 **Documentación del ejercicio por parte de la alumna**📝:`

**
**

------------------------------------------------------------------------------------------------------------------------------------------------------------------

### 🚦 Tarea 4: Filtrado por Condiciones

Ahora seleccionaremos filas que cumplan criterios específicos. Esto es muy potente para aislar datos de interés.

**Instrucciones:**
1.  Crea un DataFrame `df_alto_rend` que contenga solo las sesiones donde `Puntuacion_Rendimiento` fue mayor a 8.0. Muestra las primeras 5 filas.
2.  Crea un DataFrame `df_descanso_bajo` con las sesiones donde `Horas_Sueño_Previas` fue menor a 6.5. Muestra las primeras 5 filas.
3.  Crea un DataFrame `df_cond_combinada` con sesiones donde `Puntuacion_Rendimiento` sea mayor a 7.5 **Y** `Horas_Sueño_Previas` sea mayor o igual a 7. Muestra cuántas sesiones cumplen esto y las primeras 3 filas. Utiliza los operadores lógicos `&` (and) y `|` (or) para combinar condiciones.

In [14]:
#1

df_alto_rend = df_fenix[df_fenix['Puntuacion_Rendimiento'] > 8.0]
display(df_alto_rend.head())

ID_Patinadora  Alias_Patinadora        Fecha_Sesion            Tipo_Sesion  \
0       FENIX_11          ZigZag45 2024-08-27 19:00:00  Simulacro Competición   
3       FENIX_07  FuriaTranquila61 2024-09-16 00:00:00  Simulacro Competición   
4       FENIX_11          ZigZag45 2024-09-28 15:15:00            Desconocido   
10      FENIX_08        Vendaval51 2024-09-26 00:00:00                    NaN   
13      FENIX_09     NovaProdigy21 2024-08-16 16:30:00  Simulacro Competición   

    Duracion_Min  Pulsaciones_Max_BPM  Pulsaciones_Media_BPM  \
0         164.00               177.00                 148.00   
3            NaN                  NaN                    NaN   
4         107.00               194.00                 174.00   
10         30.00               199.00                 181.00   
13         84.00               178.00                 156.00   

    Puntuacion_Rendimiento  Nivel_Cansancio_Post  Horas_Sueño_Previas  \
0                     9.50                  5.00                  NaN   
3                     8.50                   NaN                 8.80   
4                     8.40                   NaN                  NaN   
10                    9.30                   NaN                 8.10   
13                    9.30                   NaN                 8.00   

               Comentario_Patinadora Comentario_Entrenadora  \
0                                NaN                    NaN   
3   No me sentí cómoda con las botas                    NaN   
4                                NaN                    NaN   
10  No me sentí cómoda con las botas                    NaN   
13                               NaN                     NR   

   Equipamiento_Principal_Test  Numero_Caidas_Leves  Numero_Fallos_Tecnicos  \
0             BOTAS_AIRFLOW_V1                 0.00                    1.00   
3           RUEDAS_GRIPMAX_88A                 1.00                    0.00   
4                 Ruedas Rojas                 1.00                    1.00   
10       PAT_MOD_CARBONX_R_STD                 3.00                    2.00   
13            BOTAS_AIRFLOW_V1                 0.00                    1.00   

   Rival_Observada_ID  Foco_Mental_Reportado Lesion_Reportada_S_N  \
0                 NaN                   3.00                 Leve   
3      RIVAL_LOBAS_04                   1.00                    N   
4                 NaN                   5.00                    N   
10  RIVAL_HALCONES_03                   3.00                    N   
13     RIVAL_LOBAS_03                   4.00                    N   

   Zona_Entrenamiento_Codigo  Suplementacion_Autorizada  \
0            Pista Principal  Barrita Proteica Almendra   
3            Pista Principal                    Ninguno   
4                    PISTA_A                         NR   
10           EXTERIOR_PARQUE                  Pendiente   
13                   PISTA_B                  Pendiente   

   Objetivo_Principal_Sesion  
0          Resistencia Fondo  
3        Recuperacion Activa  
4                    General  
10         Coreografia Grupo  
13              Flexibilidad

In [15]:
#2
df_descanso_bajo = df_fenix[df_fenix['Horas_Sueño_Previas'] < 6.5]
display(df_descanso_bajo.head())

ID_Patinadora Alias_Patinadora        Fecha_Sesion  \
16      FENIX_12  EstrellaFugaz39 2024-10-25 09:15:00   
49      FENIX_10         ZigZag97 2024-09-02 00:00:00   
67      FENIX_09    NovaProdigy21 2024-08-20 15:00:00   
68      FENIX_01       Dinamita27 2024-09-24 00:00:00   
69      FENIX_14  CentellaNegra76 2024-09-18 00:00:00   

                  Tipo_Sesion  Duracion_Min  Pulsaciones_Max_BPM  \
16                Comp. Local           NaN               198.00   
49      Entrenamiento General           NaN                  NaN   
67                Comp. Local        143.00               195.00   
68  Técnica Específica Saltos         34.00               174.00   
69        Rutina Coreográfica         48.00               182.00   

    Pulsaciones_Media_BPM  Puntuacion_Rendimiento  Nivel_Cansancio_Post  \
16                 182.00                    8.80                  4.00   
49                 148.00                    7.80                   NaN   
67                 180.00                   -5.00                  4.00   
68                 151.00                     NaN                  5.00   
69                    NaN                     NaN                  3.00   

    Horas_Sueño_Previas Comentario_Patinadora Comentario_Entrenadora  \
16                 6.40                   NaN                    NaN   
49                 6.40                   NaN                    NaN   
67                 6.20                   NaN                    NaN   
68                 6.00                   NaN          Revisar vídeo   
69                 6.30        Hoy todo fluyó                    NaN   

   Equipamiento_Principal_Test  Numero_Caidas_Leves  Numero_Fallos_Tecnicos  \
16            BOTAS_AIRFLOW_V1                  NaN                     NaN   
49            BOTAS_AIRFLOW_V1                 0.00                    4.00   
67      PAT_MOD_TITANIUM_R_PRO                 0.00                    0.00   
68      PAT_MOD_TITANIUM_R_PRO                 0.00                    4.00   
69      PAT_MOD_TITANIUM_R_PRO                  NaN                    0.00   

   Rival_Observada_ID  Foco_Mental_Reportado Lesion_Reportada_S_N  \
16                NaN                   4.00                    N   
49                NaN                   2.00                   No   
67                NaN                   4.00                    N   
68                NaN                    NaN                    N   
69   IRVAL_SIRENAS_01                   4.00                 Leve   

   Zona_Entrenamiento_Codigo Suplementacion_Autorizada  \
16                       YGM      Creatina Monohidrato   
49           Pista Principal                   Ninguno   
67                       GYM      Creatina Monohidrato   
68                       GYM         Isotonico Naranja   
69                   PISTA_B     Creatina Monohiddrato   

   Objetivo_Principal_Sesion  
16             Flexibillidad  
49            Velocidad Max.  
67        Tecnica Salto Axel  
68         Coreografia Grupo  
69              Flexibilidad

In [16]:
#3   Crea un DataFrame `df_cond_combinada` con sesiones donde `Puntuacion_Rendimiento` sea mayor a 7.5 **Y** `Horas_Sueño_Previas` sea mayor o igual a 7. 
# Muestra cuántas sesiones cumplen esto y las primeras 3 filas. 
# Utiliza los operadores lógicos `&` (and) y `|` (or) para combinar condiciones.T
condicion1 = df_fenix['Puntuacion_Rendimiento'] > 7.5
condicion2 = df_fenix['Horas_Sueño_Previas'] >= 7
df_cond_combinada = df_fenix[condicion1 & condicion2]
display(df_cond_combinada)

ID_Patinadora  Alias_Patinadora        Fecha_Sesion  \
3        FENIX_07  FuriaTranquila61 2024-09-16 00:00:00   
10       FENIX_08        Vendaval51 2024-09-26 00:00:00   
13       FENIX_09     NovaProdigy21 2024-08-16 16:30:00   
27       FENIX_03     AlmaDeHielo46 2024-10-25 00:00:00   
35       FENIX_03     AlmaDeHielo46 2024-08-10 00:00:00   
39       FENIX_15            Nova16 2024-09-23 12:30:00   
51       FENIX_04     Torbeellino63 2024-10-14 17:30:00   
56       FENIX_15               NaN 2024-10-24 00:00:00   
59       FENIX_07  FuriaTranquila61 2024-09-09 08:15:00   
62       FENIX_02   EstrellaFugaz77 2024-10-30 00:00:00   
64       FENIX_01        Dinamita27                 NaT   
70       FENIX_04      Torbellino63 2024-08-21 00:00:00   
80       FENIX_04      Torbellino63 2024-08-21 18:15:00   
81       FENIX_06       AceroPuro97 2024-10-15 00:00:00   
94       FENIX_06       AceroPuro97 2024-10-13 00:00:00   
96       FENIX_09     NovaProdigy21 2024-08-11 17:15:00   
106      FENIX_08        Vendaval51 2024-10-13 00:00:00   
113      FENIX_10          ZigZag97 2024-08-03 14:30:00   
117      FENIX_10          ZigZag97 2024-08-03 00:00:00   
118      FENIX_02   EstrellaFugaz77 2024-08-03 00:00:00   
128      FENIX_13       AceroPuro21 2024-10-12 13:15:00   
131      FENIX_15            Nova16 2024-09-01 18:30:00   
133      FENIX_04     Torrbellino63 2024-09-08 15:45:00   
136      FENIX_05        Dinamita88 2024-09-19 00:00:00   
148      FENIX_09     NovaProdigy21 2024-08-03 15:30:00   
150      FENIX_03     AlmaDeHielo46 2024-10-27 00:00:00   
154      FENIX_04      Torbellino63 2024-10-06 00:00:00   
158      FENIX_09     NovaProdigy21 2024-09-28 00:00:00   
174      FENIX_05        Dinamita88 2024-09-10 13:45:00   
179      FENIX_03     AlmaDeHielo46 2024-10-11 00:00:00   
182      FENIX_08        Vendaval51 2024-10-24 00:00:00   
183      FENIX_06       AceroPuro97 2024-09-02 00:00:00   
192      FENIX_06       AceroPuro97                 NaT   
195      FENIX_07  FuriaTranquila61 2024-08-25 14:30:00   
197      FENIX_06       AceroPuro97 2024-10-03 00:00:00   
198      FENIX_04      Torbellino63 2024-10-18 00:00:00   

                   Tipo_Sesion  Duracion_Min  Pulsaciones_Max_BPM  \
3        Simulacro Competición           NaN                  NaN   
10                         NaN         30.00               199.00   
13       Simulacro Competición         84.00               178.00   
27          Preparación Física           NaN               194.00   
35       Entrenamiento General         79.00               175.00   
39          Preparación Física           NaN               186.00   
51         Rutina Coreográfica         98.00               191.00   
56       Simulacro Competición           NaN               187.00   
59         Rutina Coreopráfica           NaN               161.00   
62                          NR        176.00               162.00   
64                         NaN        106.00               198.00   
70           Libre Supervisado         90.00               161.00   
80      Clasificatoria Interna        109.00               200.00   
81         Rutina Coreográfica        144.00               172.00   
94                 Comp. Local        167.00               172.00   
96          Preparación Física        142.00               187.00   
106    Técnica Específica Pies        115.00               196.00   
113                  CompLocal           NaN               161.00   
117                  CompLocal           NaN               195.00   
118        Rutina Coreográfica           NaN               188.00   
128                  Pendiente           NaN               170.00   
131          Libre Supervisado        143.00               167.00   
133        Rutina Coreográfica         36.00               192.00   
136      Entrenamiento General           NaN               166.00   
148    Técnica Específica Pies         85.00               178.00   
150  Técnica Específica Sa

------------------------------------------------------------------------------------------------------------------------------------------------------------------

`📝 **Documentación del ejercicio por parte de la alumna**📝:`

**
**

------------------------------------------------------------------------------------------------------------------------------------------------------------------

### ✨ Tarea 5: Creando Nuevas Perspectivas

Crear nuevas columnas a partir de las existentes nos ayuda a enriquecer el análisis. Se pueden crear por asignación directa.

**Instrucciones:**
1.  Crea una nueva columna `Faltas_Totales` en `df_fenix` que sea la suma de `Numero_Caidas_Leves` y `Numero_Fallos_Tecnicos`.
    * Antes de sumar, asegúrate de que los NaNs en estas dos columnas se traten como 0 para la suma, para no propagar NaNs innecesariamente si una de las dos es NaN pero la otra no.
2.  Muestra las primeras filas de `df_fenix` con esta nueva columna y las columnas originales de faltas.

In [17]:
if not df_fenix.empty:

    if 'Numero_Caidas_Leves' and 'Numero_Fallos_Tecnicos' in df_fenix.columns:
        faltas_caidas = df_fenix['Numero_Caidas_Leves'].fillna(0)
        faltas_fallos = df_fenix['Numero_Fallos_Tecnicos'].fillna(0)
        df_fenix['Faltas_Totales'] = faltas_caidas + faltas_fallos
        print("Mostrando primeras filas del DF con la nueva columna: \n\n")
        display(df_fenix[['Alias_Patinadora', 'Numero_Caidas_Leves', 'Numero_Fallos_Tecnicos', 'Faltas_Totales']].head(15))
    else:
        print("Ha ocurrido un error buscando las columnas")
else:
    print('El DF está vacío')

Mostrando primeras filas del DF con la nueva columna: 




Alias_Patinadora  Numero_Caidas_Leves  Numero_Fallos_Tecnicos  \
0           ZigZag45                 0.00                    1.00   
1         Dinamita88                 2.00                    0.00   
2      NovaProdigy21                 0.00                     NaN   
3   FuriaTranquila61                 1.00                    0.00   
4           ZigZag45                 1.00                    1.00   
5           ZigZag97                 0.00                    0.00   
6        Desconocido                  NaN                     NaN   
7         Vendaval51                 0.00                     NaN   
8        AceroPuro97                  NaN                    1.00   
9        AceroPuro21                  NaN                    2.00   
10        Vendaval51                 3.00                    2.00   
11   CentellaNegra76                 3.00                     NaN   
12       AceroPuro97                 0.00                     NaN   
13     NovaProdigy21                 0.00                    1.00   
14        Vtndaval51                 3.00                    0.00   

    Faltas_Totales  
0             1.00  
1             2.00  
2             0.00  
3             1.00  
4             2.00  
5             0.00  
6             0.00  
7             0.00  
8             1.00  
9             2.00  
10            5.00  
11            3.00  
12            0.00  
13            1.00  
14            3.00

------------------------------------------------------------------------------------------------------------------------------------------------------------------

`📝 **Documentación del ejercicio por parte de la alumna**📝:`

**
**

------------------------------------------------------------------------------------------------------------------------------------------------------------------

### 🎯 Tarea 6: Análisis Enfocado en Patinadoras Clave - Primeras Pistas del "Salseo"

Ahora, aplicaremos nuestras habilidades de filtrado para examinar más de cerca a **"EstrellaFugaz77"** y **"NovaProdigy21"**, como nos pidió "La Centella".

**Instrucciones:**
1.  Filtra `df_fenix` para obtener todos los registros de "EstrellaFugaz77" y guárdalos en `df_estrella`.
2.  Sobre `df_estrella`, filtra las sesiones de `Tipo_Sesion` que sean "Simulacro Competición" o "Clasificatoria Interna". Calcula la `Puntuacion_Rendimiento` media de "EstrellaFugaz77" en estas sesiones.
3.  Repite los puntos 1 y 2 para "NovaProdigy21", guardando en `df_nova` y calculando su media.
4.  Compara las medias de ambas.
5.  Muestra algunos comentarios únicos (no nulos y no vacíos) de `Comentario_Patinadora` y `Comentario_Entrenadora` para "EstrellaFugaz77".
6.  Haz lo mismo para "NovaProdigy21".

In [18]:
if not df_fenix.empty and 'Alias_Patinadora' in df_fenix:

    alias_estrella = "EstrellaFugaz77"
    alias_nova = "NovaProdigy21"
    tipos_sesion_filtro = ['Simulacro Competición', 'Clasificatoria Interna']

#Analisis para Estrella Fugaz 77:
    df_estrella = df_fenix[df_fenix['Alias_Patinadora'] == alias_estrella].copy()

    if not df_estrella.empty:
        sesion_estrella = df_estrella[df_estrella['Tipo_Sesion'].isin(tipos_sesion_filtro)]
        rendimiento_medio_estrella = sesion_estrella['Puntuacion_Rendimiento'].mean()
        print(f"\n Analisis de {alias_estrella} \n:")
        print(f"Puntuación media en sesiones clave es {rendimiento_medio_estrella}, basado en {len(sesion_estrella)} sesiones")
        if "Comentario_Patinadora" in df_estrella.columns:
            print("ALgunos comentarios de Estrella77:")
            print(df_estrella[df_estrella['Comentario_Patinadora'].notna() & (df_estrella['Comentario_Patinadora'] != "")]['Comentario_Patinadora'].unique()[:3])
        if "Comentario_Entrenadora" in df_estrella.columns:
            print("ALgunos comentarios de Estrella:")
            print(df_estrella[df_estrella['Comentario_Entrenadora'].notna() & (df_estrella['Comentario_Entrenadora'] != "")]['Comentario_Entrenadora'].unique()[:3])
    else:
        print(f"No se encontraron datos para {alias_estrella}")

#Analisis para Nova Prodigy 21:
    df_nova = df_fenix[df_fenix['Alias_Patinadora'] == alias_nova].copy()

    if not df_nova.empty:
        sesion_nova = df_nova[df_nova['Tipo_Sesion'].isin(tipos_sesion_filtro)]
        rendimiento_medio_nova = sesion_nova['Puntuacion_Rendimiento'].mean()
        print(f"\n Analisis de {alias_nova} \n:")
        print(f"Puntuación media en sesiones clave es {rendimiento_medio_nova}, basado en {len(sesion_nova)} sesiones")
        if "Comentario_Patinadora" in df_nova.columns:
            print("ALgunos comentarios de Nova21:")
            print(df_nova[df_nova['Comentario_Patinadora'].notna() & (df_nova['Comentario_Patinadora'] != "")]['Comentario_Patinadora'].unique()[:3])
        if "Comentario_Entrenadora" in df_nova.columns:
            print("ALgunos comentarios de Nova:")
            print(df_nova[df_nova['Comentario_Entrenadora'].notna() & (df_nova['Comentario_Entrenadora'] != "")]['Comentario_Entrenadora'].unique()[:3])
    else:
        print(f"No se encontraron datos para {alias_nova}")


 Analisis de EstrellaFugaz77 
:
Puntuación media en sesiones clave es nan, basado en 1 sesiones
ALgunos comentarios de Estrella77:
['NR' 'No me sentí cómoda con las botas' 'Presionada por NovaProdigy21']
ALgunos comentarios de Estrella:
['Necesita más trabajo de base' 'NR' 'Actitud impecable']

 Analisis de NovaProdigy21 
:
Puntuación media en sesiones clave es 9.3, basado en 2 sesiones
ALgunos comentarios de Nova21:
['Necesito mejorar mi Salto Axel' 'Ruedas nuevas van genial'
 'Molestia tobillo izq.']
ALgunos comentarios de Nova:
['Excelente progresión en giros' 'NR' 'Falta explosividad']


In [19]:
# Ante la falta de datos basandonos en los 2 tipo de sesión especificado, he decidido hacer una media general de todas las sesiones para poder compararlos.

print(f"En en analisis total de NovaProdigy21 hemos observado una puntuación media de {df_nova['Puntuacion_Rendimiento'].mean():.2f} sobre {len(df_nova['Puntuacion_Rendimiento'])} sesiones con datos")
print("-"*100)
print(f"En en analisis total de EstrellaFugaz77 hemos observado una puntuación media de {df_estrella['Puntuacion_Rendimiento'].mean():.2f} sobre {len(df_estrella['Puntuacion_Rendimiento'])} sesiones con datos")




En en analisis total de NovaProdigy21 hemos observado una puntuación media de 8.02 sobre 13 sesiones con datos
----------------------------------------------------------------------------------------------------
En en analisis total de EstrellaFugaz77 hemos observado una puntuación media de 7.43 sobre 11 sesiones con datos


In [20]:
# Aprovecho que hemos creado los DF de cada patinadora, desarrollo el ejercicio voluntario con la media de faltas totales de cada una de ellas.

print(f"En en analisis de faltas de NovaProdigy21 hemos observado una puntuación media de {df_nova['Faltas_Totales'].mean():.2f} sobre {len(df_nova['Faltas_Totales'])} sesiones con datos")
print("-"*100)
print(f"En en analisis total de EstrellaFugaz77 hemos observado una puntuación media de {df_estrella['Faltas_Totales'].mean():.2f} sobre {len(df_estrella['Faltas_Totales'])} sesiones con datos")

En en analisis de faltas de NovaProdigy21 hemos observado una puntuación media de 0.62 sobre 13 sesiones con datos
----------------------------------------------------------------------------------------------------
En en analisis total de EstrellaFugaz77 hemos observado una puntuación media de 1.36 sobre 11 sesiones con datos


------------------------------------------------------------------------------------------------------------------------------------------------------------------

`📝 **Documentación del ejercicio por parte de la alumna**📝:`

**
**

------------------------------------------------------------------------------------------------------------------------------------------------------------------

### 📄 Tarea 7: Primer Informe de Inteligencia para "La Centella"

**Resumen de Limpieza y Transformación:**
* 
    *Carga del CSV `rendimiento_fenix.csv`. 
    *Realizada inspección general. Convertida la columna `Fecha_Sesion` a formato datetime. 
    *Columnas numéricas relevantes fueron procesadas para asegurar tipo numérico, manejando errores de conversión con `coerce`.
    *Se creó la columna `Faltas_Totales`.

**Observaciones Iniciales sobre "EstrellaFugaz77" y "NovaProdigy21":**

* **Rendimiento Medio en Sesiones Clave (Simulacro Competición / Clasificatoria Interna):**
    * "EstrellaFugaz77": `*No hemos logrado una media de rendimiento en las columnas solicitadas, no hay datos suficientes*`
    * "NovaProdigy21": `*Presenta una media de 9,3 basada en las dos sesiones que hemos podido revisar dentro de las columnas solicitadas*`
    * *Comentario de la alumna: ¿Hay una diferencia notable a simple vista? ¿Alguna es más consistente o con mejor rendimiento en estas sesiones de presión?*
        `Según los datos que nos solicita la señora Centella, destaca el rendimiento de NovaProdigy21 siendo sobresaliente y además la única de ambas de la que disponemos la información solicitada. Al ver que los datos se quedaban un tanto "pobres" hemos decidido ampliar la búsqueda y ofrecemos aquí también la media de las columnas totales de rendimientos con datos, por si le pudiera esclarecer un poco más la situación.`

        `En en analisis total de NovaProdigy21 hemos observado una puntuación media de 8.02 sobre 13 sesiones con datos`

        `En en analisis total de EstrellaFugaz77 hemos observado una puntuación media de 7.43 sobre 11 sesiones con datos`

        `Ante esta forma de revisión, NovaProdigy21 sigue destacando más en rendimiento, aunque podemos observar que el de EstrellaFugaz77 también es notable`

* **Media de Faltas Totales (General):**
    * "EstrellaFugaz77": `Hemos observado una puntuación media de 1.36 sobre 11 sesiones con datos`

    * "NovaProdigy21": `En en analisis de faltas de NovaProdigy21 hemos observado una puntuación media de 0.62 sobre 13 sesiones con datos`

    * *Comentario de la alumna: ¿Alguna parece cometer significativamente más faltas?* `La media de fallos de EstrellaFugaz77 dobla, e incluso un poco más, a las faltas de NovaProdigy21`

* **Comentarios Relevantes Encontrados:**
    * Para "EstrellaFugaz77": `Los comentarios mas relevantes de la patinadora han sido 'No me sentí cómoda con las botas' y 'Presionada por NovaProdigy21', y de la entrenadora resulta interesante que considera que 'Necesita más trabajo de base' pero recalca su 'actitud impecable'`

    * Para "NovaProdigy21": `En el caso de Nova destacamos que comenta que tiene molestias en el tobillo izquierdo y que necesita mejorar su salto axel, lo que concuerda perfectamente con el comentario de la Centella de que necesita más "explosividad". Al mismo tiempo notamos que la Centella ha notado una mejoría en sus giros y Nova comenta que sus ruedas nuevas vam genial, a lo que también encontramos una relación lógica`


* **Próximas Preguntas/Sospechas para Investigar:**
    * `Considero importante profundizar en la falta de datos de EstrellaFugaz sobre el tipo de sesión solicitadas, para ver porque se están registrando mal, ya que destrozan su dato de rendimiento y como hemos visto, aplicando una media general realmente su rendimiento no está mal. También destacaría intentar entender el comentario que hace sobre su compañera, por si está habiendo una rivalidad que le hace faltar o cometer más errores, cosa que el equipo no necesita, ya que se nombra que tiene una actitud impecable y es extraño que se le considere así cuando se queja abiertamente de una compañera`

        `Por otro lado, Nova parece una gran promesa con una gran puntuació y pocas faltas, pero debemos averiguar si lo del tobillo puede jugarnos malas pasadas, quizá fue algo momentaneo pero quizá a la larga le perjudique como patinadora y a nosotras como equipo`


### 💾 Guardado del DataFrame Procesado

Guardaremos nuestro DataFrame `df_fenix` (con la limpieza de tipos básicos y la nueva columna `Faltas_Totales`) para usarlo en el siguiente Notebook de EDA.

**Instrucción:**
* Guarda `df_fenix` como `rendimiento_fenix_pandas_limpio.csv`.

In [21]:
# Guardar el DataFrame para el siguiente notebook

if not df_fenix.empty:
    try:
        df_fenix.to_csv('rendimiento_fenix_pandas_limpio.csv', sep=';', index=False, encoding='utf-8-sig')
        print("\nDataFrame guardado exitosamente como 'rendimiento_fenix_pandas_limpio.csv'")
    except Exception as e:
        print(f"Error al guardar el DataFrame: {e}")
else:
    print("El DataFrame está vacío, no se guardó.")


DataFrame guardado exitosamente como 'rendimiento_fenix_pandas_limpio.csv'


### ✅ Misión 02: ¡Primeras Pistas y Datos Organizados! 🐼💪

¡Excelente trabajo con Pandas, analista!

Has organizado los datos de "Las Fénix" y, lo más importante, has empezado a tirar del hilo de algunas dinámicas que "La Centella" tenía entre ceja y ceja. 

Ahora ella puede:

* Ver un resumen claro del estado del equipo y los tipos de datos que manejamos.
* Seleccionar rápidamente la información de patinadoras o sesiones específicas usando `loc` e `iloc`.
* Filtrar datos para encontrar justo lo que necesita, incluyendo las primeras comparativas entre "EstrellaFugaz77" y "NovaProdigy21".
* Analizar nuevas métricas como `Faltas_Totales`.

Estos son los cimientos, y las primeras observaciones sobre nuestras patinadoras clave ya nos dan mucho en qué pensar. 

En la próxima misión (EDA), profundizaremos en estos patrones y buscaremos más respuestas.


🔥🔥🔥🔥🔥🔥🔥🔥🔥🔥🔥🔥🔥🔥